tests/epyccel/test_arrays.py

```diff
@@ -3975,6 +3975,808 @@ def test_array_ndmin_2_order(language):
+
+@pytest.mark.parametrize( 'language', (
+        pytest.param("fortran", marks = [
+            pytest.mark.skip(reason=("Should fail until type conversion is supported between non compatible types, see #1785")),
+            pytest.mark.fortran]),
+        pytest.param("c", marks = pytest.mark.c),
+        pytest.param("python", marks = pytest.mark.python)
+    )
+)
+def test_dtype_conversion_int64_to_other_types_4(language):
+    size = (2, 2)
+    integer64 = randint(low = iinfo('int64').min, high = iinfo('int64').max , size = size, dtype=np.int64)
+
+    epyccel_func = epyccel(arrays.dtype_convert_to_pyint, language=language)
+    assert epyccel_func(integer64) == arrays.dtype_convert_to_pyint(integer64)
+
+    epyccel_func = epyccel(arrays.dtype_convert_to_pyfloat, language=language)
+    assert epyccel_func(integer64) == arrays.dtype_convert_to_pyfloat(integer64)
+
+@pytest.mark.parametrize( 'language', (
+        pytest.param("fortran", marks = [
+            pytest.mark.skip(reason=("Should fail until type conversion is supported between non compatible types, see #1785")),
+            pytest.mark.fortran]),
+        pytest.param("c", marks = pytest.mark.c),
+        pytest.param("python", marks = pytest.mark.python)
+    )
+)
+def test_dtype_conversion_float32_to_other_types_1(language):
+    size = (2, 2)
+    fl32 = uniform(finfo('float32').min / 2, finfo('float32').max / 2, size = size)
+    fl32 = np.float32(fl32)
+
+    epyccel_func = epyccel(arrays.dtype_convert_to_int8, language=language)
+    assert epyccel_func(fl32) == arrays.dtype_convert_to_int8(fl32)
+
+    epyccel_func = epyccel(arrays.dtype_convert_to_bool, language=language)
+    assert epyccel_func(fl32) == arrays.dtype_convert_to_bool(fl32)
+
+    epyccel_func = epyccel(arrays.dtype_convert_to_int16, language=language)
+    assert epyccel_func(fl32) == arrays.dtype_convert_to_int16(fl32)
+
+@pytest.mark.parametrize( 'language', (
+        pytest.param("fortran", marks = [
+            pytest.mark.skip(reason=("Should fail until type conversion is supported between non compatible types, see #1785")),
+            pytest.mark.fortran]),
+        pytest.param("c", marks = pytest.mark.c),
+        pytest.param("python", marks = pytest.mark.python)
+    )
+)
+def test_dtype_conversion_float32_to_other_types_2(language):
+    size = (2, 2)
+    fl32 = uniform(finfo('float32').min / 2, finfo('float32').max / 2, size = size)
+    fl32 = np.float32(fl32)
+
+    epyccel_func = epyccel(arrays.dtype_convert_to_int32, language=language)
+    assert epyccel_func(fl32) == arrays.dtype_convert_to_int32(fl32)
+
+    epyccel_func = epyccel(arrays.dtype_convert_to_int64, language=language)
+    assert epyccel_func(fl32) == arrays.dtype_convert_to_int64(fl32)
+
+    epyccel_func = epyccel(arrays.dtype_convert_to_float32, language=language)
+    assert epyccel_func(fl32) == arrays.dtype_convert_to_float32(fl32)
```

Comment:
The error that you reported is about casting to a logical (boolean) type. Not all of these tests require such a conversion. Are you sure that they all fail?

Reply:
Skipping the test cases for fortran language only for cases it will result in compilation error. Other cases, fortran tests are run now.